### PR TITLE
chore: pause on input in pwdebug mode

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/server/supplements/injected/selectorGenerator.ts [hunk 1]
```diff
@@ -26,6 +26,19 @@ type SelectorToken = {
 const cacheAllowText = new Map<Element, SelectorToken[] | null>();
 const cacheDisallowText = new Map<Element, SelectorToken[] | null>();
 
+export function querySelector(injectedScript: InjectedScript, selector: string, ownerDocument: Document): { selector: string, elements: Element[] } {
+  injectedScript._evaluator.begin();
```
Comment (dgozman) [inline src/server/supplements/injected/selectorGenerator.ts:30]: You don't have to begin/end wrap this, querySelectorAll does it for you.

File: src/server/supplements/recorderSupplement.ts [hunk 2]
```diff
@@ -294,4 +315,50 @@ export class RecorderSupplement {
...
+
+  private _source(metadata: CallMetadata): { source: string, line: number | undefined } {
+    let source = '// No source available';
+    let line: number | undefined = undefined;
+    if (metadata.stack && metadata.stack.length) {
+      try {
+        source = this._readAndCacheSource(metadata.stack[0].file);
+        line = metadata.stack[0].line ? metadata.stack[0].line - 1 : undefined;
+      } catch (e) {
+        source = metadata.stack.join('\n');
```
Comment (dgozman) [inline src/server/supplements/recorderSupplement.ts:346]: Stack is an array of structs, not strings, isn't it?

File: src/utils/stackTrace.ts [hunk 3]
```diff
@@ -50,6 +50,8 @@ export function captureStackTrace(): { stack: string, frames: StackFrame[] } {
     // for tests.
     if (fileName.includes(path.join('playwright', 'src')))
       continue;
+    if (fileName.includes(path.join('playwright', 'test', 'coverage.js')))
```
Comment (dgozman) [inline src/utils/stackTrace.ts:53]: Seems sketchy. Perhaps only when isUnderTest?

File: src/web/recorder/recorder.tsx [hunk 4]
```diff
@@ -21,12 +21,15 @@ import { ToolbarButton } from '../components/toolbarButton';
 import { Source } from '../components/source';
 
 type Mode = 'inspecting' | 'recording' | 'none';
-
+type PauseDetails = {
```
Comment (dgozman) [inline src/web/recorder/recorder.tsx:24]: Can this include recorderTypes?

**Review by dgozman — APPROVED** [review]

File: src/server/supplements/recorderSupplement.ts (hunk 2 above)
Comment (pavelfeldman) [reply src/server/supplements/recorderSupplement.ts:346]: Will follow up!

**Pull request merged**